content/200-concepts/100-components/08-prisma-engines/200-query-engine.mdx

```diff
@@ -10,36 +10,36 @@ From a technical perspective, Prisma Client consists of three major components:
 
 - JavaScript client library
 - TypeScript type definitions
-- A query engine (in the form of binary file)
+- A query engine
 
 All of these components are located in the [generated `.prisma/client` folder](../prisma-client/working-with-prismaclient/generating-prisma-client#the-prismaclient-npm-package) after you ran `prisma generate`.
 
 This page covers relevant technical details about the query engine.
 
 </TopBlock>
 
-## The query engine binary file
+## The query engine file
 
-The **query engine binary file** is called `query-engine-PLATFORM` where `PLATFORM` corresponds to the name of a compile target. For example, if the query engine must run on a [Darwin](<https://en.wikipedia.org/wiki/Darwin_(operating_system)>) operating system (e.g. Mac OS), it's called `query-engine-darwin`. You can find an overview of all supported platforms [here](../../../reference/api-reference/prisma-schema-reference#binarytargets-options).
+The **query engine file** is different for each operating system. It is named `query-engine-PLATFORM` or `libquery_engine-PLATFORM` where `PLATFORM` corresponds to the name of a compile target, file extensions vary and are dependent on the platform as well. For example, if the query engine must run on a [Darwin](<https://en.wikipedia.org/wiki/Darwin_(operating_system)>) operating system (e.g. Mac OS), it's called `libquery_engine-darwin.dylib.node` or `query-engine-darwin`. You can find an overview of all supported platforms [here](../../../reference/api-reference/prisma-schema-reference#binarytargets-options).
 
-The query engine binary file is downloaded into the `runtime` directory of `node_modules/@prisma/client` when `prisma generate` is called.
+The query engine file is downloaded into the `runtime` directory of the generated Prisma Client when `prisma generate` is called.
 
 Note that the query engine is implemented in Rust. The source code is located in the [`prisma-engines`](https://github.com/prisma/prisma-engines/) repository.
 
 ## The query engine at runtime
 
-The query engine is running as a separate process on the same machine where the Node.js/TypeScript application is running.
+The query engine is a [Node-API library](https://nodejs.org/api/n-api.html) that is loaded into the Prisma Client library. (There is also is also another engine type for the query engine that is compiled to an executable binary, that is run as a sidecar process alongside your application. Read more about [configuring the query engine](#configuring-the-query-engine). Compared to the binary method, the Node-API library uses a newer technique for binding Prisma's Rust-based query engine directly to Prisma Client. This reduces the communication overhead between the Node.js and Rust layers when resolving Prisma Client's database queries, so is usually beneficial to use.)
```

Comment:
```suggestion
The query engine is a [Node-API library](https://nodejs.org/api/n-api.html) that is loaded into the Prisma Client library. (There is also another engine type for the query engine that is compiled to an executable binary, that is run as a sidecar process alongside your application. Read more about [configuring the query engine](#configuring-the-query-engine). Compared to the binary method, the Node-API library uses a newer technique for binding Prisma's Rust-based query engine directly to Prisma Client. This reduces the communication overhead between the Node.js and Rust layers when resolving Prisma Client's database queries, so is usually beneficial to use.)
```

Reply:
I'd also recommend cutting down repetition of "that is" in this phrase.

Reply:
```suggestion
By default, the Prisma Client loads the query engine as a [Node-API library](https://nodejs.org/api/n-api.html). You can alternatively [configure Prisma to use the query engine compiled as an executable binary](#configuring-the-query-engine), which is run as a sidecar process alongside your application. 
The Node-API library approach is recommended since it reduces the communication overhead between the Prisma Client and the query engine.
```
Tried to simplify a bit here